include/fastdds/rtps/common/Time_t.h

```diff
@@ -372,28 +380,55 @@ inline std::istream& operator >>(
 
     if (s)
     {
+        // Variable to store point in number
         char point;
+        // Variable for seconds
         int32_t sec = 0;
+        // In this string the nano number will be stored. It is needed to store str to check it has 9 digits.
+        std::string nano_st;
+        // Variable for nanoseconds
         uint32_t nano = 0;
         std::ios_base::iostate excp_mask = input.exceptions();
 
         try
         {
             input.exceptions(excp_mask | std::ios_base::failbit | std::ios_base::badbit);
 
-            input >> sec;
-            input >> point >> nano;
-            // nano could not be bigger than 1 sec
-            if ( point != '.' || nano > 1000000000 )
+            input >> sec; // Get seconds value
+            input >> point >> nano_st; // Get point and nanoseconds string value
+            if (point != '.')
             {
-                input.setstate(std::ios_base::failbit);
+                // Only seconds
                 nano = 0;
             }
+            else
+            {
+                nano = static_cast<uint32_t>(std::stoul(nano_st));
+
+                if (nano > 1000000000ULL)
+                {
+                    // Error case, nanos > 10^9
+                    input.setstate(std::ios_base::failbit);
```

Comment:
Why not truncate to the 9 most significant digits? Also, this is not being tested. Please add a test for this case.